### ISSUE TITLE
[Question] Some questions about the PPO_lag algorithm

### ISSUE THREAD

**Issue by tjruan** [PKU-Alignment/omnisafe#348]
### Required prerequisites

- [X] I have read the documentation <https://omnisafe.readthedocs.io>.
- [X] I have searched the [Issue Tracker](https://github.com/PKU-Alignment/omnisafe/issues) and [Discussions](https://github.com/PKU-Alignment/omnisafe/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [X] Consider asking first in a [Discussion](https://github.com/PKU-Alignment/omnisafe/discussions/new).

### Questions

Hello Omnisafe team, thank you very much for your contribution. I am experiencing some confusion and I hope you can answer it for me, I appreciate it! 
The original PPO algorithm uses the CLIP objective function approach. In the documentation it is mentioned that the surrogate loss function for the PPOlag algorithm is：
<img width="446" alt="Snipaste_2024-08-16_12-12-44" src="https://github.com/user-attachments/assets/739cc6c5-0718-4150-b668-ca03f7657357">
Does this equation represent an advantage function that combines rewards and costs?
Is this L in PPOlag replacing A(s, a) in PPO?
<img width="418" alt="Snipaste_2024-08-16_12-18-35" src="https://github.com/user-attachments/assets/4e0c1a09-bf5b-41f0-afac-a7abc615cc7e">
If so, can you point me to where in the code the clip is in PPOlag?
I would greatly appreciate it if you could answer my questions.

**Comment by Gaiejj** [comment]
> Does this equation represent an advantage function that combines rewards and costs?

Yes, this is an objective function that considers both reward advantage and cost advantage simultaneously.

> Is this L in PPOlag replacing A(s, a) in PPO?

Sure, it is right.

> If so, can you point me to where in the code the clip is in PPOlag?

The implementation of PPOLag simply replaces \( A(s, a) \) with the objective function weighted by the Lagrange multiplier. Therefore, the clipping operation can be found in the PPO implementation of omnisafe, specifically in lines 69 to 76 of the loss calculation function in PPO:
```python
        ratio = torch.exp(logp_ - logp)
        ratio_cliped = torch.clamp(
            ratio,
            1 - self._cfgs.algo_cfgs.clip,
            1 + self._cfgs.algo_cfgs.clip,
        )
        loss = -torch.min(ratio * adv, ratio_cliped * adv).mean()
        loss -= self._cfgs.algo_cfgs.entropy_coef * distribution.entropy().mean()
```
